slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Importer.java

```diff
@@ -0,0 +1,115 @@
+import tech.pegasys.web3signer.slashingprotection.dao.SignedBlocksDao;
+import tech.pegasys.web3signer.slashingprotection.dao.Validator;
+import tech.pegasys.web3signer.slashingprotection.dao.ValidatorsDao;
+import tech.pegasys.web3signer.slashingprotection.interchange.model.Metadata;
+import tech.pegasys.web3signer.slashingprotection.interchange.model.SignedAttestation;
+import tech.pegasys.web3signer.slashingprotection.interchange.model.SignedBlock;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.tuweni.bytes.Bytes;
+import org.jdbi.v3.core.Jdbi;
+
+public class InterchangeV5Importer {
+
+  private static final int FORMAT_VERSION = 5;
+
+  private final Jdbi jdbi;
+  private final ValidatorsDao validatorsDao;
+  private final SignedBlocksDao signedBlocksDao;
+  private final SignedAttestationsDao signedAttestationsDao;
+  private final ObjectMapper mapper;
+
+  public InterchangeV5Importer(
+      final Jdbi jdbi,
+      final ValidatorsDao validatorsDao,
+      final SignedBlocksDao signedBlocksDao,
+      final SignedAttestationsDao signedAttestationsDao,
+      final ObjectMapper mapper) {
+    this.jdbi = jdbi;
+    this.validatorsDao = validatorsDao;
+    this.signedBlocksDao = signedBlocksDao;
+    this.signedAttestationsDao = signedAttestationsDao;
+    this.mapper = mapper;
+  }
+
+  public void importData(final InputStream input) throws IOException {
+
+    try (JsonParser jsonParser = mapper.getFactory().createParser(input)) {
+      final ObjectNode rootNode = mapper.readTree(jsonParser);
+
+      final JsonNode metadataJsonNode = rootNode.get("metadata");
+      final Metadata metadata = mapper.treeToValue(metadataJsonNode, Metadata.class);
+
+      if (metadata.getFormatVersion() != FORMAT_VERSION) {
+        throw new IllegalStateException("Expecting an interchange_format_version of 5");
+      }
+
+      final JsonNode dataJsonNode = rootNode.get("data");
+      final ArrayNode dataNode = (ArrayNode) dataJsonNode;
+      for (int i = 0; i < dataNode.size(); i++) {
+        final JsonNode validatorNode = dataNode.get(i);
+        parseValidator(validatorNode);
+      }
+    }
+  }
+
+  private void parseValidator(final JsonNode node) throws JsonProcessingException {
+    if (node.getNodeType() != JsonNodeType.OBJECT) {
+      throw new IllegalStateException("Element of 'data' was not an object");
+    }
+    final ObjectNode parentNode = (ObjectNode) node;
+    final String pubKey = parentNode.get("pubkey").textValue();
+    final Validator validator =
```

Comment:
Probably want these all to be in the same transaction? Otherwise will only rollback part of the import if it fails. Is that what is intended?

Reply:
Yeah, this is a big change, and need to work out how to do it.